src/datasources/cache/redis.cache.service.ts

```diff
@@ -46,6 +46,12 @@ export class RedisCacheService implements ICacheService, OnModuleDestroy {
     return await this.client.unlink(cacheDir.key);
   }
 
+  async deleteByPattern(pattern: string): Promise<void> {
+    for await (const key of this.client.scanIterator({ MATCH: pattern })) {
+      await this.client.unlink(key);
+    }
+  }
```

Comment:
We should really avoid performing a scan over a pattern. One of the biggest selling features of the rewrite is that the way we interact with Redis should be more deterministic. Therefore if we need to delete all the "values" for a specific feature we should think of a solution that groups those values.

Therefore we should strive to call `UNLINK` to the key that holds these values. If we need to change the key structure for that then we should evaluate that first.

Reply:
Good point! We could change the key in the case of tokens and contracts (which are the entities being deleted by pattern in this branch) and use the Redis hash to store just the reference to the chain in the _key_ part, and the concrete token/contract address in the _field_ part. So that will allow us to delete all the contracts/tokens on a chain in one call to Redis. Do you see any downside to this approach?